terminatorlib/plugins/save_last_session_layout.py

```diff
@@ -1,82 +1,123 @@
+
+        # OS signal handlers for shutdown scenarios
+        signal.signal(signal.SIGTERM, self.signal_handler)
+        signal.signal(signal.SIGHUP, self.signal_handler)
 
-    #not used, but capability can be used to load automatically
-    def load_session_layout(self, debugtab=False, widget=None, cwd=None, metadata=None, profile=None):
-      dbg("SaveLastSessionLayout load layout")
-      terminator = Terminator()
-      util.spawn_new_terminator(terminator.origcwd, ['-u', '-l', 'SaveLastSessionLayout'])
+        terminator = Terminator()
 
-    def save_session_layout(self, debugtab=False, widget=None, cwd=None, metadata=None, profile=None):
+        # Connect to all existing terminals
+        for terminal in terminator.terminals:
+            self.connect_to_terminal(terminal)
 
-      config = Config()
-      terminator = Terminator()
-      current_layout = terminator.describe_layout(save_cwd = True)
-      dbg("SaveLastSessionLayout: save layout(%s)" % current_layout)
-      res = config.replace_layout("SaveLastSessionLayout", current_layout)
-      if (not res):
-        r = config.add_layout("SaveLastSessionLayout", current_layout)
-      config.save()
-      return True
+        # Connect to all existing notebooks for new tab detection
+        self.connect_to_notebooks()
 
-    def signal_handler(self,signum, frame):
+        return False  # Don't repeat - run only once
 
+    # Not used, but capability can be used to load automatically
+    def load_session_layout(self, debugtab=False, widget=None, cwd=None, metadata=None, profile=None):
+        """Load the saved session layout in a new Terminator instance."""
+        dbg("SaveLastSessionLayout: load layout")
+        terminator = Terminator()
+        util.spawn_new_terminator(terminator.origcwd, ['-u', '-l', 'SaveLastSessionLayout'])
+
+    def save_session_layout(self):
+        """Save the current layout to config."""
+        config = Config()
+        terminator = Terminator()
+        current_layout = terminator.describe_layout(save_cwd=True)
+        dbg("SaveLastSessionLayout: save layout(%s)" % current_layout)
+        res = config.replace_layout("SaveLastSessionLayout", current_layout)
+        if not res:
+            config.add_layout("SaveLastSessionLayout", current_layout)
+        config.save()
+        return True
+
+    def signal_handler(self, signum, frame):
+        """Handle OS signals (SIGTERM, SIGHUP)."""
         signame = signal.Signals(signum).name
-        dbg('signal handler called:signal %s (%s)' %
-                                        (signame, signum))
+        dbg('SaveLastSessionLayout: signal handler called: %s (%s)' % (signame, signum))
         self.save_session_layout()
-    
-    def connect_signals(self):
-        dbg("SaveLastSessionLayout connect_signals")
-
-        signal.signal(signal.SIGTERM, self.signal_handler)
-        signal.signal(signal.SIGCHLD, self.signal_handler)
```

Comment:
This seemed like a noisy signal that we might not need as things inside open terminals are terminated